operators/online_video_reasoner/tests/test_online_video_reasoner.py

```
@@ -0,0 +1,812 @@
+        app.reasoner._frames[0], np.full((12, 16, 3), 173, dtype=np.uint8)
+    )
+
+
+@pytest.mark.parametrize("missing_stream", [None, 0])
+def test_cuda_input_without_stream_uses_default_copy(reasoning_server, monkeypatch, missing_stream):
+    operator = _operator(reasoning_server, mode="image", stream=False)
+    expected = np.arange(12 * 16 * 3, dtype=np.uint8).reshape(12, 16, 3)
+    calls = []
+
+    class CudaTensor:
+        __cuda_array_interface__ = {}
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)